polaris-service/src/main/java/org/apache/polaris/service/config/PolarisApplicationConfig.java

```diff
@@ -208,4 +215,58 @@ public void setAwsSecretKey(String awsSecretKey) {
   public void setDefaultRealms(List<String> defaultRealms) {
     this.defaultRealms = defaultRealms;
   }
+
+  public Supplier<GoogleCredentials> getGcpCredentialsProvider() {
+    return gcpAccessToken != null
+        ? () -> GoogleCredentials.create(gcpAccessToken)
+        : () -> {
+          try {
+            return GoogleCredentials.getApplicationDefault();
+          } catch (IOException e) {
+            throw new RuntimeException("Failed to get GCP credentials", e);
+          }
+        };
+  }
+
+  @JsonProperty("gcp_credentials")
+  void setGcpCredentials(GcpAccessToken token) {
+    this.gcpAccessToken =
+        new AccessToken(
+            token.getAccessToken(),
+            new Date(System.currentTimeMillis() + token.getExpiresIn() * 1000));
```

Comment:
Should we reduce the duration slightly to guard against skew?

Reply:
There's not really a concern about making this accurate. There's no way to refresh this currently and users are never expected to pass in an access token via configuration string. This really only exists to support testing and the token passed in is never used.